### PR TITLE
Login: New Jetpack required screen

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #8075

### Description

This PR adds a few UI changes to prepare for the native Jetpack setup flow. The design can be found here: ERGtCFLyiC9CFpHsw1XIid-fi-1110%3A37319. 

Changes include:
- New temporary feature flag for the native Jetpack setup flow. This should be replaced with an A/B test later.
- Handle navigation to the native Jetpack setup flow when the user enters a site address in the site discovery flow (from the store picker) and the site doesn't have Jetpack or is not connected to the user's WP.com account.
- Add a new screen for the Jetpack error reusable for both Jetpack installation and connection.
- Also: update the empty store picker screen following the new design.

### Testing instructions

Prerequisite: Make sure that you have a test store that is not associated with your WP.com account or doesn't have Jetpack.

- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- Notice that you are navigated to a new Jetpack required screen matching the Figma design. Ensure that the copy and CTA is correct based on the state of Jetpack on your site.

### Screenshots

| Jetpack not installed or activated | Jetpack connection missing |
| ----- | ----- |
|  |  |

Empty store picker:
| Before | After |
| ----- | ----- |
|  |  |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.swift [hunk 1]
```diff
@@ -68,7 +77,9 @@ private extension EmptyStoresTableViewCell {
             comment: "Link on the store picker for users who signed in with Apple to close their WordPress.com account."
         )
         static let legend =
-            NSLocalizedString("Your account isn’t connected to any WooCommerce stores.",
+            NSLocalizedString("Create your first store.",
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.swift]: Super nit: Do we need a full stop here? I don't see one on the Figma design.

File: WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.xib [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="21179.7" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="21507" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.xib:2]: Super nit: There seems to be a background color difference when dark mode is turned on.

File: WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            comment: "Button to install Jetpack from the Jetpack setup required screen"
+        )
+        static let connectJetpack = NSLocalizedString(
+            "Connect Jetpack",
+            comment: "Button to authorize Jetpack connection from the Jetpack setup required screen"
+        )
+        static let setupErrorMessage = NSLocalizedString(
+            "To use this app for %@ you'll need the free Jetpack plugin installed and connected on your store.",
+            comment: "Error message on the Jetpack setup required screen." +
+            "Reads like: To use this app for test.com you'll need...")
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift]: Super nit: Consider moving `)` to a new line for consistency.

File: WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        static let connectionErrorMessage = NSLocalizedString(
+            "To use this app for %@ you'll need to connect your store to Jetpack.",
+            comment: "Error message on the Jetpack setup required screen when Jetpack connection is missing." +
+            "Reads like: To use this app for test.com you'll need...")
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift]: Super nit: Consider moving `)` to a new line for consistency.

File: WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        )
+        static let connectionSubtitle = NSLocalizedString(
+            "You’ll need your store credentials to begin the connection.",
+            comment: "Subtitle on the Jetpack setup required screen when only Jetpack connection is missing"
+        )
+        static let helpBarButtonItemTitle = NSLocalizedString("Help", comment: "Help button on Jetpack setup required screen.")
+        static let termsContent = NSLocalizedString(
+            "By tapping the Connect Jetpack button, you agree to our %1$@ and to %2$@ with WordPress.com.",
+            comment: "Content of the label at the end of the Jetpack setup required screen. " +
+            "Reads like: By tapping the Connect Jetpack button, you agree to our Terms of Service and to share details with WordPress.com.")
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift]: Super nit: Consider moving `)` to a new line for consistency.

File: WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            "Terms of Service",
+            comment: "The terms to be agreed upon when tapping the Connect Jetpack button on the Jetpack setup required screen."
+        )
+        static let shareDetails = NSLocalizedString(
+            "share details",
+            comment: "The action to be agreed upon when tapping the Connect Jetpack button on the Jetpack setup required screen."
+        )
+    }
+
+    enum Strings {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupRequiredViewModel.swift]: Super nit: This consists of URLs, and maybe a name like `Links` or `Constants` will suit this better?

File: WooCommerce/WooCommerceTests/Authentication/JetpackSetupRequiredViewModelTests.swift [hunk 7]
```diff
@@ -0,0 +1,173 @@
+import XCTest
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/Authentication/JetpackSetupRequiredViewModelTests.swift:1]: Nice tests. ❤️

**Review by selanthiraiyan — APPROVED** [review]
It works as expected. Nice tests. 🎉 🚢 

I left a few non-blocking super nits.

File: WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.swift]: Good catch! Removed in b60842af0646be698ff14d675e463d6130c27656.

File: WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.xib (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.xib:2]: Awesome, I missed this! Fixed in d14063de27a0b350786e38a39fe4c9f3c1dd06af.

**Pull request merged**